### ISSUE TITLE
Handle the initwithlocale parameter

### ISSUE THREAD

**Issue by pietro909** [RobertSheaO/react-native-zendesk-support#11]
On Zendesk docs the is the possibility to use 
[initwithlocale](https://developer.zendesk.com/embeddables/docs/ios/help_center#initwithlocale) in order to use documents with different translations.

This is not exposed here though, do you have any idea how to implement it?

Thanks!

**Comment by RobertSheaO** [comment]
So far this package doesn't implement any of the API Providers APIs. I'll look into initializing the built in help center in a different locale and provide a patch for that. As for the rest of those APIs, they're on the roadmap for integration, but require a lot of extra work.

**Comment by pietro909** [comment]
Thank you RobertSheaO! Please let me know if I can help somehow.

**Comment by RobertSheaO** [comment]
@pietro909 I've added `initWithLocale` support; however, I'm unable to test, as I don't have an enterprise Zendesk account that allows me to specify multiple locales. Can you test to make sure it works? I put it in the initWithLocale branch. You can reference it from your `package.json` like so:

`"react-native-zendesk-support": "git+https://github.com/RobertSheaO/react-native-zendesk-support#initWithLocale"`

You can specify a locale on any Help Center method by passing "locale" as an optional param.

e.g. `ZendeskSupport.showHelpCenterWithOptions({ locale: 'zh-tw' })`

Please report back your results and I'll pull this into the master branch and cut a release.

P.S. The ZendeskSDK doesn't support passing locale for Android, it uses the device locale and defaults to `en-us` otherwise.

**Comment by RobertSheaO** [comment]
@pietro909 I see you gave a thumbs up, did specifying a locale work with the initWithLocale branch?

**Comment by pietro909** [comment]
No, sorry. It was to show that I got the info and I'm going to plan this test in our schedule :-)
I think Friday (tomorrow) will be a good day for this.

**Comment by RobertSheaO** [comment]
Any update on this working for you? I’m ready to merge to master once you confirm it works in another locale.

**Comment by pietro909** [comment]
Hi Robert, sorry for the delay. Doesn't seem to work though.

I'm calling it with:

`showHelpCenterWithOptions({ locale: "de" })`

And I have German content. Nonetheless, it loads always the English version.
Here's a debugger's view to show that the parameter is being passed to `showHelpCenterWithOptions` handler in ObjectiveC:

![screen shot 2017-12-01 at 16 30 34](https://user-images.githubusercontent.com/2094604/33489948-ca212876-d6b5-11e7-9cd4-9ffce883e34f.png)

**Comment by RobertSheaO** [comment]
Hi! Just saw you posted an edit. That's great that the locale param is passing through properly; however, you're still seeing English content be returned to the app?

**Comment by RobertSheaO** [comment]
Hello again @pietro909. I've just merged the latest from master into the initWithLocale branch. Can you confirm if the code is working for you, or if there is an update on your end about debugging it? From your last message, it certainly sounded like the code made a request for the correct content, yet you still got English content?

**Comment by pietro909** [comment]
Hi Robert, apparently the problem is the one you described. Although our Zendesk configuration looks good.
I need to schedule some time to debug and go through it, it's good that you merged it in any case: maybe someone else will start using that feature and help shed some light on my issue.

I will get back on this later!

**Comment by RobertSheaO** [comment]
I need to write the Android implementation and then I'll merge it into master

**Comment by mhonig** [comment]
Hi Robert. Is there any progress on this one? Thanks, Moritz
